src/javascript/events/component-view.js

```diff
@@ -0,0 +1,86 @@
+/*global module, require */
+'use strict'; // eslint-disable-line strict
+
+const Core = require('../core');
+const getTrace = require('../../libs/get-trace');
+const utils = require('../utils');
+
+const TRACKING_ATTRIBUTES = [
+	'contentId',
+	'type',
+	'subtype',
+	'component',
+];
+
+const decorateEventData = (eventData, viewedEl, opts) => {
+
+	const { trace, customContext } = getTrace(viewedEl);
+	let context;
+
+	if (opts.getContextData) {
+		if (typeof opts.getContextData !== 'function') {
+			throw new Error('opts.getContextData is not a function');
+		}
+
+		const dataBeforeWhitelist = opts.getContextData(viewedEl);
+
+		if (typeof dataBeforeWhitelist !== 'object') {
+			throw new Error('opts.getContextData function should return {Object}');
+		}
+
+		context = utils.whitelistProps(dataBeforeWhitelist, TRACKING_ATTRIBUTES);
+	} else {
+		context = {};
+	}
+
+	context.domPathTokens = trace;
+	context.url = window.document.location.href || null;
+
+	utils.assignIfUndefined(customContext, context);
+
+	eventData.context = context;
+};
+
+/**
+ * Listen for view events.
+ *
+ * @alias view#init
+ * @param {Object} opts - To set custom category[String], selector[String], getContextData[Function]
+ * @return {undefined}
+ */
+const init = (opts = {}) => {
+	if(!window.IntersectionObserver) {
```

Comment:
It's good that this is optional to avoid a breaking change :) But perhaps a warning here would be good to help debug, if users expect to track views across older browsers?
```
console.warn('o-tracking: Unable to track element view events as "window.IntersectionObserver" is not supported.');
```

Reply:
👍